test_values/test_cases.json

```diff
@@ -252,5 +252,196 @@
+        "e52d9c508c502347344d8c07ad91cbd6068afc75ff6292f062a09ca381c89e71",
+        "e7cf46a078fed4fafd0b5e3aff144802b853f8ae459a4f0c14add3314b7cc3a6",
+        "9576f4ade6e9bc3a6458b506ce3e4e890df29cb14cb5d3d887672aef55647a2b",
+        "34028bbc87000c39476cdc60cf80ca32d579b3a0e2d3f80e0ad8c3739a01aa91",
+        "cd9c77062a338e63a63ca623db438cb8676f15466641079ee61ec2dda98de796"
+      ]
+    },
+    {
+      "leaf_values": [
+        0, 1, 2, 3, 4, 5, 6, 7, 
+        8, 9, 10, 11, 12, 13, 14
+      ],
+      "target_values": [0, 1, 10, 5],
+      "proof_hashes": [
+        "e52d9c508c502347344d8c07ad91cbd6068afc75ff6292f062a09ca381c89e71",
+        "e7cf46a078fed4fafd0b5e3aff144802b853f8ae459a4f0c14add3314b7cc3a6",
+        "9576f4ade6e9bc3a6458b506ce3e4e890df29cb14cb5d3d887672aef55647a2b",
+        "34028bbc87000c39476cdc60cf80ca32d579b3a0e2d3f80e0ad8c3739a01aa91",
+        "cd9c77062a338e63a63ca623db438cb8676f15466641079ee61ec2dda98de796"
+      ],
+      "roots": [
+        "352d0d5e172316e73b6d0d40605ad411c79b7e7c1ed0b4529c565f08057bc4a8",
+        "cae921bbf649d4dd84c252c55c540e2e30c6c00cb089d9704bd613ecea308643",
+        "55d0a0ef8f5c25a9da266b36c0c5f4b31008ece82df2512c8966bddcc27a66a0",
+        "4d7b3ef7300acf70c892d8327db8272f54434adbc61a4e130a563cb59a0d0f47"
+      ]
+    },
+    {
+      "leaf_values": [
+        0, 1, 2, 3, 4, 5, 6, 7, 
+        8, 9, 10, 11, 12, 13, 14
+      ],
+      "target_values": [0, 1, 5, 14],      
+      "proof_hashes": [
+        "e52d9c508c502347344d8c07ad91cbd6068afc75ff6292f062a09ca381c89e71",
+        "9576f4ade6e9bc3a6458b506ce3e4e890df29cb14cb5d3d887672aef55647a2b",
+        "34028bbc87000c39476cdc60cf80ca32d579b3a0e2d3f80e0ad8c3739a01aa91"
+      ],
+      "roots":[
+        "352d0d5e172316e73b6d0d40605ad411c79b7e7c1ed0b4529c565f08057bc4a8",
+        "9c053db406c1a077112189469a3aca0573d3481bef09fa3d2eda3304d7d44be8",
+        "55d0a0ef8f5c25a9da266b36c0c5f4b31008ece82df2512c8966bddcc27a66a0",
+        "0000000000000000000000000000000000000000000000000000000000000000"
+      ] 
+    },
+    {
+      "leaf_values": [
+        0, 1, 2, 3, 4, 5, 6, 7, 
+        8, 9, 10, 11, 12, 13, 14
+      ],
+      "target_values": [1, 6, 5, 3],
+      "proof_hashes": [
+        "6e340b9cffb37a989ca544e6bb780a2c78901d3fb33738768511a30617afa01d",
+        "dbc1b4c900ffe48d575b5da5c638040125f65db0fe3e24494b76ea986457d986",
+        "e52d9c508c502347344d8c07ad91cbd6068afc75ff6292f062a09ca381c89e71",
+        "ca358758f6d27e6cf45272937977a748fd88391db679ceda7dc7bf1f005ee879"
+      ],
+      "roots": [
+        "d6ab1f11dd0ceee1862aa84b5e4d2c7520a8651fa2652529e8c49cc2b43b4476",
+        "9c053db406c1a077112189469a3aca0573d3481bef09fa3d2eda3304d7d44be8",
+        "55d0a0ef8f5c25a9da266b36c0c5f4b31008ece82df2512c8966bddcc27a66a0",
+        "4d7b3ef7300acf70c892d8327db8272f54434adbc61a4e130a563cb59a0d0f47"
+      ]
+    },
+    {
+      "leaf_values": [
```

Comment:
Nit: Same for here. leaf_values for the test case below have different formatting but they're the same thing. Having the same formatting would be good